src/v2/Apps/Onboarding/useOnboardingContext.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  FC,
+  createContext,
+  createRef,
+  useCallback,
+  useRef,
+  useContext,
+  useState,
+  Dispatch,
+  SetStateAction,
+} from "react"
+import { WorkflowEngine } from "v2/Utils/WorkflowEngine"
+import { useEngine } from "./config"
+
+export type Context = {
+  answer: string | null
+}
+
+export const DEFAULT_CONTEXT: Context = {
+  answer: null,
+}
+
+type AnswersState = [] | [string] | [string, string]
+
+const OnboardingContext = createContext<{
+  answers: AnswersState
+  context: React.RefObject<Context>
+  current: string
+  engine: WorkflowEngine
+  next(): void
+  onDone(): void
+  progress: number
+  setAnswers: Dispatch<SetStateAction<AnswersState>>
+  setContext: (updatedContext: Partial<Context>) => React.RefObject<Context>
+}>({
+  answers: [],
+  context: createRef<Context>(),
+  current: "",
+  engine: new WorkflowEngine({ workflow: [] }),
+  next: () => {},
+  onDone: () => {},
+  progress: 0,
+  setContext: () => createRef<Context>(),
```

Comment:
Ooh, maybe we literally are setting a new react context? 

Still, i wonder how this can be made less opaque as part of the decision engine and / or onboarding flow. Name-wise its too low-level of a thing to have to be called within individual steps, but if it was named differently it might not feel like so

Reply:
No, I agree. It's a little weird / should be renamed.